compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -4436,6 +4436,43 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
         }
     }
 
+    /// If the type failed selection but the trait is implemented for `(T,)`, suggest that the user
+    /// creates a unary tuple
+    ///
+    /// This is a common gotcha when using libraries that emulate variadic functions with traits for tuples.
+    pub(super) fn suggest_tuple_wrapping(
+        &self,
+        err: &mut Diag<'_>,
+        root_obligation: &PredicateObligation<'tcx>,
+        obligation: &PredicateObligation<'tcx>,
+        span: Span,
+    ) {
+        if !matches!(obligation.cause.code(), ObligationCauseCode::FunctionArg { .. }) {
```

Comment:
I would turn this into an `if let ObligationCauseCode::FunctionArg { arg_hir_id, .. } = ...`.

Then use the span from `arg_hir_id` via `tcx.hir.span(arg_hir_id)`. It may seem roundabout, but there's always a chance that the obligation span has been messed with, and getting the span from the actual arg expression is strictly more right...